src/metrics/metadata/metadata.go

```diff
@@ -273,13 +273,18 @@ func (metadatas PipelineMetadatas) ApplyOrRemoveDropPolicies() (
 		return metadatas, NoDropPolicyPresentResult
 	}
 
+	result := metadatas
+
+	// Drop is effective as no other non drop pipelines, result is a drop
 	if nonDropPipelines == 0 {
-		// Drop is effective as no other non drop pipelines, result is a drop
-		return DropPipelineMetadatas, AppliedEffectiveDropPolicyResult
+		// nb: Do not directly return DropPipelineMetadatas, as the client could potentially save that reference
+		// and modify global state.
+		result = result[:0]
+		result = append(result, DropPipelineMetadata)
```

Comment:
seems bad to return `AppliedEffectiveDropPolicyResult` as well, somebody could in theory alter that too (even though it's not happening today)

Reply:
i think we should refactor all of these global vars into accessor methods that return copies of package protected vars.